api/preprint_providers/views.py

```diff
@@ -0,0 +1,189 @@
+from rest_framework import generics
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.models import Node, PreprintProvider
+
+from api.base import permissions as base_permissions
+from api.base.filters import ODMFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.base.pagination import MaxSizePagination
+
+from api.preprint_providers.serializers import PreprintProviderSerializer
+from api.preprints.serializers import PreprintSerializer
+
+
+class PreprintProviderMixin(object):
+    institution_lookup_url_kwarg = 'institution_id'
```

Comment:
This should probably be `preprintprovider_lookup_url_kwarg = 'preprintprovider_id'`